src/fetch-or-populate-file-cache.ts

```diff
@@ -0,0 +1,77 @@
+import { fileExists, readJsonFile, writeJsonFile } from '@metamask/utils/node';
+import type { Json } from '@metamask/utils/node';
+
+import { ONE_HOUR } from './constants';
+import { createModuleLogger, projectLogger } from './logging-utils';
+
+const log = createModuleLogger(projectLogger, 'fetch-or-populate-file-cache');
+
+/**
+ * The data stored in the cache file.
+ */
+type FileCache<Data extends Json> = {
+  /**
+   * When the data was stored.
+   */
+  ctime: string;
+  /**
+   * The cached data.
+   */
+  data: Data;
+};
+
+/**
+ * How long to cache data retrieved from an API (to prevent rate limiting).
+ *
+ * Equal to 1 hour.
+ */
+const DEFAULT_MAX_AGE = ONE_HOUR;
+
+/**
+ * Avoids rate limits when making requests to an API by consulting a file cache.
+ *
+ * Reads the given cache file and returns the data within it if it exists and is
+ * fresh enough; otherwise runs the given function and saves its return value to
+ * the file.
+ *
+ * @param args - The arguments to this function.
+ * @param args.filePath - The path to the file where the data should be saved.
```

Comment:
Nit: Perhaps we could update these docs, or the parameter name (or both) to make it more clear that this is the path to the _cache file_? When I first read this, I thought this was how the data was output (e.g. that this was a "download data and write to this file" function.